### PR TITLE
Queries for unknown and aborted statuses

### PR DESCRIPTION
**Pull request by mgcam**

### CODE REVIEW

File: lang_qc/db/helper/wells.py [hunk 1]
```diff
@@ -147,12 +147,23 @@ class PacBioPagedWellsFactory(PagedStatusResponse):
         description="A SQLAlchemy Session for the ml warehouse database",
     )
 
+    # For MySQL it's OK ot use case-sensitive comparison operators since
```
Comment (nerdstrike) [inline lang_qc/db/helper/wells.py:150]: ot to you too

File: lang_qc/db/helper/wells.py [hunk 2]
```diff
@@ -147,12 +147,23 @@ class PacBioPagedWellsFactory(PagedStatusResponse):
...
+    # its string comparisons are string-insensitive.
```
Comment (nerdstrike) [inline lang_qc/db/helper/wells.py:151]: subject to collation. Ours is `utf8mb4_0900_ai_ci` which is mercifully insensitive.

File: lang_qc/db/helper/wells.py [hunk 3]
```diff
@@ -147,12 +147,23 @@ class PacBioPagedWellsFactory(PagedStatusResponse):
...
     FILTERS: ClassVar = {
         QcFlowStatusEnum.ON_HOLD.name: (QcStateDict.state == "On hold"),
         QcFlowStatusEnum.QC_COMPLETE.name: (QcState.is_preliminary == 0),
         QcFlowStatusEnum.IN_PROGRESS.name: and_(
             QcState.is_preliminary == 1, QcStateDict.state != "On hold"
         ),
+        QcFlowStatusEnum.ABORTED.name: or_(
+            PacBioRunWellMetrics.well_status.like("Abort%"),
```
Comment (nerdstrike) [inline lang_qc/db/helper/wells.py:159]: Does the warehouse data need us to be this flexible? It smells bad (and has performance impact).

File: lang_qc/db/helper/wells.py [hunk 4]
```diff
@@ -167,16 +178,23 @@ def create(self) -> PacBioPagedWells:
...
 
         wells = []
         if self.qc_flow_status == QcFlowStatusEnum.INBOX:
             recent_wells = WellWh(session=self.mlwh_session).recent_completed_wells()
             wells = self._recent_inbox_wells(recent_wells)
+        elif self.qc_flow_status in [
+            QcFlowStatusEnum.ABORTED,
+            QcFlowStatusEnum.UNKNOWN,
+        ]:
+            wells = self._aborted_and_unknown_wells()
```
Comment (nerdstrike) [inline lang_qc/db/helper/wells.py:197]: Aha, perhaps you could change L191 to match this `self` invocation? It seems to silly to instantiate a new one in this context.

**Review by nerdstrike — CHANGES_REQUESTED** [review]
Me likey. Comments and typo only.

File: lang_qc/db/helper/wells.py (hunk 4 above)
Comment (mgcam) [reply lang_qc/db/helper/wells.py:197]: On line 91 an object of a different class is instantiated. This object (self) does not inherit from it. Might be it should.

Comment (mgcam) [reply lang_qc/db/helper/wells.py:196]: Now it does, see https://github.com/wtsi-npg/npg_langqc/pull/110/commits/5f053f9f77e07b053e67230247cdfaab8c604c24

**Review by nerdstrike — APPROVED** [review]

**Pull request merged**